### DISCUSSION TITLE
Is there any way to embed a asciinema in a README.md?

### DISCUSSION THREAD

**Discussion by Unitech** [asciinema/discussions#284] [Q&A]
I tried to embed the script into the README of PM2 (https://github.com/Unitech/PM2) without success. I think Github avoid inserting scripts to avoid XSS attacks, but maybe there is another way to do it ?

Thanks for asciinema !

**Comment by ku1ik** [comment]
There's a plan to generate snaphot PNG from recordings so they can be inserted in readmes (as markdown image links for example). Please follow asciinema/asciinema-server#152

**Comment by ku1ik** [comment]
FYI there's "image link" (PNG) embedding now, as well as support for oEmbed/Open Graph/Twitter Card: https://asciinema.org/docs/embedding

**Comment by BenGitsCode** [comment]
I know this is closed, but I'd like to pop in a [README](https://github.com/PizzaBeer/filebucket-front-end/blob/master/README.md) I did before discovering the gem that is ASCII cinema. I recorded those gifs using Gyazo and embedded them in md the same way you could embed any ASCII cinema link.

**Comment by markvanderloo** [comment]
@sickill Nice info but that doesn't answer the original question (to me). Is it possible, and if so, how?

**Comment by ku1ik** [comment]
@markvanderloo in short: it's not possible to embed the player in README (Github doesn't allow script execution), but you can embed PNG preview which links to the recording on asciinema.org. See https://docs.asciinema.org/manual/server/embedding/

**Comment by markvanderloo** [comment]
Thanks!

**Comment by themightychris** [comment]
@sickill @BenGitsCode @markvanderloo check out https://github.com/nbedos/termtosvg

Maybe asciinema.org could make use of it behind the scenes to offer embed/download options for uploaded recordings?

**Comment by ku1ik** [comment]
@themightychris thanks for bringing this up. Yeah, we could integrate it in some way into the site, or find a good way to collaborate with devs of termtosvg, to make this easier for users.

**Comment by tniessen** [comment]
You can use https://github.com/marionebl/svg-term-cli to convert the cast file to an animated SVG and then embed it, similar to how termtosvg worked.

**Comment by DanielBiegler** [comment]
I am aware that this is a closed issue but I just got here through a search engine so I'll contribute in case others arrive here as well.

### How to make it work in 2025

Run your recording session with the v2 format instead of the newer v3 because `svg-term-cli` only supports v2

```bash
asciinema rec -f asciicast-v2  # ... your other args
# or convert existing animation to v2 format
asciinema convert -f asciicast-v2  # ... your other flags
```

Use the here mentioned `svg-term-cli` to generate an svg:

```bash
npx svg-term-cli < usage-demo.json > usage-demo.svg  # ... your other flags
```

It works :)

![Image](https://github.com/user-attachments/assets/9c849000-953d-4258-ac94-f7444251c5f3)
